wger/core/views/user.py

```diff
@@ -323,6 +330,82 @@ def preferences(request):
         return render(request, 'user/preferences.html', template_data)
 
 
+@login_required
+def sync_fitbit(request):
+    """Authorizes the connection with Wger
+
+    """
+
+    if settings.WGER_SETTINGS['FITBIT_CLIENT_ID'] is None:
+        raise Http404("Fitbit not configured correctly to sync")
+
+    fitbit_id = settings.WGER_SETTINGS['FITBIT_CLIENT_ID']
+    fitbit_secretkey = settings.WGER_SETTINGS['FITBIT_CLIENT_SECRET']
+
+    # Implement the OAuth2 authorization flow
+    fitbit_auth_client = fitbit.FitbitOauth2Client(fitbit_id, fitbit_secretkey)
+
+    weight_template = {}
+
+    if 'code' in request.GET:
+        code = request.GET.get('code', '')
+
+        form = {
+            'client_secret': fitbit_secretkey,
+            'code': code,
+            'client_id': fitbit_id,
+            'grant_type': 'authorization_code',
+            'redirect_uri': os.getenv('ROOT_URL')+'/en/user/sync_fitbit/'
+        }
+
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            "Authorization": 'Basic ' + base64.b64encode(
+                fitbit_id.encode() + b':' + fitbit_secretkey.encode()).decode()
+        }
+
+        response = requests.post(
+            fitbit_auth_client.request_token_url, form, headers=headers).json()
+
+        if "access_token" in response:
+            token = response['access_token']
+            headers['Authorization'] = 'Bearer ' + token
+            date = (datetime.date.today()).strftime('%Y-%m-%d')
+
+            response = requests.get(
+                'https://api.fitbit.com/1/user/-/body/log/weight/date/' + date + '.json',
```

Comment:
Is this uri hardcoded or is it customized for every user that logs into wger?

Reply:
> Is this uri hardcoded or is it customized for every user that logs into wger?

It varies for each user. I think a lot has been abstracted by Fitbit. It seems that once you authorize Wger to get Fitbit data, that uri will somehow be custom to your account only.